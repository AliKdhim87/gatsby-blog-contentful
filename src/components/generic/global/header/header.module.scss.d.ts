export const headerStyle: string;
export const title: string;
export const navLinks: string;
export const navItem: string;
export const activeNavItem: string;
export const background: string;
