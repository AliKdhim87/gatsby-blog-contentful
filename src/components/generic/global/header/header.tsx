import React from "react"
import { Link, graphql, useStaticQuery } from "gatsby"

import {
  navItem,
  headerStyle,
  title,
  navLinks,
  activeNavItem,
} from "./header.module.scss"

const Header: React.FC = () => {
  const {
    site: {
      siteMetadata: { title: mainTitle },
    },
  } = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
          author
        }
      }
    }
  `)

  return (
    <header className={headerStyle}>
      <h1>
        <Link className={title} to="/">
          {mainTitle}
        </Link>
      </h1>
      <nav>
        <ul className={navLinks}>
          <li>
            <Link className={navItem} activeClassName={activeNavItem} to="/">
              Home
            </Link>
          </li>
          <li>
            <Link
              className={navItem}
              activeClassName={activeNavItem}
              to="/blog"
            >
              Blog
            </Link>
          </li>
          <li>
            <Link
              className={navItem}
              activeClassName={activeNavItem}
              to="/contact"
            >
              Contact
            </Link>
          </li>
          <li>
            <Link
              className={navItem}
              activeClassName={activeNavItem}
              to="/about"
            >
              About
            </Link>
          </li>
        </ul>
      </nav>
    </header>
  )
}

export default Header
