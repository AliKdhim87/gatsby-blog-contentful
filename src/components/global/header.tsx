import React, { useState } from "react"
import { useLocation } from "@reach/router"
import { graphql, useStaticQuery } from "gatsby"
import Link from "gatsby-link"

import {
  Menu,
  Container,
  Header,
  Icon,
  MenuItem,
  Sidebar,
} from "semantic-ui-react"

import { useCurrentWidth } from "hooks/useWidth"

const HeaderComponent: React.FC = () => {
  const width = useCurrentWidth()
  const { pathname } = useLocation()
  const [mobileMode, setMobileMode] = useState<boolean>(false)

  const activeLink = (path: string) => path === pathname
  const {
    site: {
      siteMetadata: { title: mainTitle },
    },
  } = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
          author
        }
      }
    }
  `)

  const navLinks = (
    <>
      <Menu.Item
        name="home"
        active={activeLink("/")}
        as={Link}
        to="/"
        onClick={mobileMode ? () => setMobileMode(false) : undefined}
      >
        Home
      </Menu.Item>
      <Menu.Item
        name="blog"
        active={activeLink("/blog")}
        as={Link}
        to="/blog"
        onClick={mobileMode ? () => setMobileMode(false) : undefined}
      >
        Blogs
      </Menu.Item>
      <Menu.Item
        name="contact"
        active={activeLink("/contact")}
        as={Link}
        to="/contact"
        onClick={mobileMode ? () => setMobileMode(false) : undefined}
      >
        Contact
      </Menu.Item>
      <Menu.Item
        name="about"
        active={activeLink("/about")}
        as={Link}
        to="/about"
        onClick={mobileMode ? () => setMobileMode(false) : undefined}
      >
        About
      </Menu.Item>
    </>
  )

  return (
    <>
      <Menu
        inverted
        secondary
        color="blue"
        size="massive"
        as="header"
        style={{ marginBottom: "0" }}
      >
        <Menu.Item>
          <Header as="span" inverted>
            {mainTitle}
          </Header>
        </Menu.Item>
        <Container text>
          {width <= 767 ? (
            <MenuItem
              style={{ padding: "1.8rem 1rem", marginRight: "0" }}
              position="right"
              visible={width <= 767 ? "visible" : "hidden"}
              onClick={() => setMobileMode(!mobileMode)}
            >
              <Icon
                name={mobileMode ? "close" : "bars"}
                inverted
                size="large"
              />
            </MenuItem>
          ) : (
            <MenuItem position="right">{navLinks}</MenuItem>
          )}
        </Container>
      </Menu>
      <Sidebar
        as={Menu}
        animation="push"
        icon="labeled"
        inverted
        onHide={() => setMobileMode(false)}
        vertical
        visible={mobileMode}
        width="wide"
      >
        {navLinks}
      </Sidebar>
    </>
  )
}

export default HeaderComponent
