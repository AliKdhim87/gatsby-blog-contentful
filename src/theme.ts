import {DefaultTheme} from 'styled-components'

export enum ColorPalette {
  background = '#F3F4F5',
  blue = '#0E6EB8',
  red = '#FF546D',
  grey = '#f8f9fa',
  black = '#1B1C1D',
  white = '#FCFCFC',
  orange = '#FFAA50',
  darkBlack = '#000',
}

const baseTheme = {
  breakpoint: {
    mobile: '767px',
  },
  palette: {
    background: ColorPalette.background,
    blue: ColorPalette.blue,
    red: ColorPalette.red,
    grey: ColorPalette.grey,
    black: ColorPalette.black,
    white: ColorPalette.white,
    orange: ColorPalette.orange,
    darkBlack: ColorPalette.darkBlack,
  },
}

export const lightTheme: DefaultTheme = {
  ...baseTheme,
  textColor: ColorPalette.black,
  mainBg: ColorPalette.grey,
  secondary: ColorPalette.red,
  primary: ColorPalette.red,
  mainBgDark: ColorPalette.grey,
}
export const darkTheme: DefaultTheme = {
  ...baseTheme,
  textColor: ColorPalette.grey,
  mainBg: ColorPalette.black,
  secondary: ColorPalette.orange,
  primary: ColorPalette.orange,
  mainBgDark: ColorPalette.darkBlack,
}
