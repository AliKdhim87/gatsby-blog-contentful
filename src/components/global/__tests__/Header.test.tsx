import React from 'react'
import renderer from 'react-test-renderer'

import {lightTheme, darkTheme} from 'theme'

import {PureHeader as Header, links} from '../header'

jest.mock(
  'semantic-ui-react/dist/commonjs/behaviors/Visibility/Visibility',
  () => ({children}: {children: React.ReactNode}) => children,
)
jest.mock(
  'semantic-ui-react/dist/commonjs/modules/Sidebar/Sidebar',
  () => ({children}: {children: React.ReactNode}) => children,
)

jest.mock('@reach/router', () => ({
  useLocation: jest.fn().mockReturnValue({pathname: '/'}),
}))

describe('Header', () => {
  it('renders correctly and select the dark mode', () => {
    const tree = renderer
      .create(
        <Header
          brandName="ALI DEV"
          links={links}
          isDark={true}
          secondary={darkTheme.secondary}
          textColor={darkTheme.textColor}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly and select the light mode', () => {
    const tree = renderer
      .create(
        <Header
          brandName="ALI DEV"
          links={links}
          isDark={false}
          secondary={lightTheme.secondary}
          textColor={lightTheme.textColor}
        />,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
