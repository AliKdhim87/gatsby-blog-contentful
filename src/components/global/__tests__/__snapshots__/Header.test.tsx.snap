// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly and select the dark mode 1`] = `
Array [
  <nav
    className="ui small inverted segment"
  >
    <div
      className="ui container"
    >
      <div
        className="ui grid"
      >
        <div
          className="row"
        >
          <a
            className="four wide computer eight wide mobile six wide tablet column"
            href="/"
          >
            <test-file-stub
              aria-label="Ali dev logo"
            />
             
            <span
              className="ui large header"
              style={
                Object {
                  "color": "#FFAA50",
                }
              }
            >
              ALI DEV
            </span>
          </a>
          <div
            className="computer only tablet only right floated middle aligned ten wide column"
          >
            <div
              className="sc-gsTCUz jUYWqf"
            >
              <a
                className="sc-bdfBwQ hjchiG active"
                href="/"
              >
                Home
              </a>
              <a
                className="sc-bdfBwQ hjchiG"
                href="/blog/"
              >
                Blog
              </a>
              <a
                className="sc-bdfBwQ hjchiG"
                href="/contact/"
              >
                Contact
              </a>
              <a
                className="sc-bdfBwQ hjchiG"
                href="/about/"
              >
                About
              </a>
              <i
                aria-hidden="true"
                className="grey moon large link icon"
                onClick={[Function]}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              />
            </div>
          </div>
          <div
            className="mobile only middle aligned eight wide column"
            onClick={[Function]}
          >
            <div
              className="ui tiny right aligned header"
            >
              <i
                aria-label="menu button"
                className="bars small inverted link icon"
                onClick={[Function]}
                style={
                  Object {
                    "color": "#f8f9fa",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="sc-gsTCUz jUYWqf"
  >
    <a
      className="sc-bdfBwQ hjchiG active"
      href="/"
    >
      Home
    </a>
    <a
      className="sc-bdfBwQ hjchiG"
      href="/blog/"
    >
      Blog
    </a>
    <a
      className="sc-bdfBwQ hjchiG"
      href="/contact/"
    >
      Contact
    </a>
    <a
      className="sc-bdfBwQ hjchiG"
      href="/about/"
    >
      About
    </a>
    <i
      aria-hidden="true"
      className="grey moon large link icon"
      onClick={[Function]}
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
  </div>,
]
`;

exports[`Header renders correctly and select the light mode 1`] = `
Array [
  <nav
    className="ui small segment"
  >
    <div
      className="ui container"
    >
      <div
        className="ui grid"
      >
        <div
          className="row"
        >
          <a
            className="four wide computer eight wide mobile six wide tablet column"
            href="/"
          >
            <test-file-stub
              aria-label="Ali dev logo"
            />
             
            <span
              className="ui large header"
              style={
                Object {
                  "color": "#FF546D",
                }
              }
            >
              ALI DEV
            </span>
          </a>
          <div
            className="computer only tablet only right floated middle aligned ten wide column"
          >
            <div
              className="sc-gsTCUz jUYWqf"
            >
              <a
                className="sc-bdfBwQ hjchiG active"
                href="/"
              >
                Home
              </a>
              <a
                className="sc-bdfBwQ hjchiG"
                href="/blog/"
              >
                Blog
              </a>
              <a
                className="sc-bdfBwQ hjchiG"
                href="/contact/"
              >
                Contact
              </a>
              <a
                className="sc-bdfBwQ hjchiG"
                href="/about/"
              >
                About
              </a>
              <i
                aria-hidden="true"
                className="red sun large link icon"
                onClick={[Function]}
                style={
                  Object {
                    "display": "inline-block",
                  }
                }
              />
            </div>
          </div>
          <div
            className="mobile only middle aligned eight wide column"
            onClick={[Function]}
          >
            <div
              className="ui tiny right aligned header"
            >
              <i
                aria-label="menu button"
                className="bars small inverted link icon"
                onClick={[Function]}
                style={
                  Object {
                    "color": "#1B1C1D",
                  }
                }
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </nav>,
  <div
    className="sc-gsTCUz jUYWqf"
  >
    <a
      className="sc-bdfBwQ hjchiG active"
      href="/"
    >
      Home
    </a>
    <a
      className="sc-bdfBwQ hjchiG"
      href="/blog/"
    >
      Blog
    </a>
    <a
      className="sc-bdfBwQ hjchiG"
      href="/contact/"
    >
      Contact
    </a>
    <a
      className="sc-bdfBwQ hjchiG"
      href="/about/"
    >
      About
    </a>
    <i
      aria-hidden="true"
      className="red sun large link icon"
      onClick={[Function]}
      style={
        Object {
          "display": "inline-block",
        }
      }
    />
  </div>,
]
`;
